back/node_watcher/src/util/getReferendumStatus.ts

```diff
@@ -0,0 +1,31 @@
+import { Option } from '@polkadot/types';
+import {
+  ReferendumInfo,
+  ReferendumInfoTo239,
+  ReferendumStatus,
+} from '@polkadot/types/interfaces';
+
+function isOld(
+  info: ReferendumInfo | ReferendumInfoTo239
+): info is ReferendumInfoTo239 {
+  return !!(info as ReferendumInfoTo239).proposalHash;
+}
+
+export function getReferendumStatus(
```

Comment:
Actually, I would also add a comment on this function, and say that it's a copy-paste from api-derive